### DISCUSSION TITLE
Inside VPN: "WebSocket connection to 'ws://172.28.224.2:8000/socket.io/?EIO=3&transport=websocket' failed:"

### DISCUSSION THREAD

**Discussion by anthnyprschka** [miguelgrinberg/Flask-SocketIO#1683] [Q&A] [answered]
**Describe the bug**
I am trying to run my Flask-SocketIO server and clients inside a VPN (OpenVPN). Unforunately I am getting this error message in the browser console:

`WebSocket connection to 'ws://172.28.224.2:8000/socket.io/?EIO=3&transport=websocket' failed:`

We have verified that 172.28.224 is indeed the IP of the machine the server is running on.

**Expected behavior**
Websocket connection can be established. Note it works fine with setting http://127.0.0.1:8000/ as URI in client and opening client in browser on _same_ machine, or deploying server on Heroku without VPN, setting Heroku link as URI and opening client from anywhere.

**Comment by miguelgrinberg** [comment, marked as answer]
Is the request reaching the server? If not, then something that you have in between the client and the server is at fault. If the requests are reaching the server, then please provide logs.

**Reply by anthnyprschka** [reply]
Requests aren't reaching the server, I think it's related to gunicorn (I'm running the flask server with gunicorn)

**Reply by anthnyprschka** [reply]
Adding `--forwarded-allow-ips="172.28.224.2"` to gunicorn command has no effect

**Reply by anthnyprschka** [reply]
Adding `-b 0.0.0.0:8000` solved it...
